src/main/resources/assets/js/components/AppModalComponent.jsx

```diff
@@ -31,17 +33,49 @@ var AppModalComponent = React.createClass({
     rollBackApp: React.PropTypes.func.isRequired,
     scaleApp: React.PropTypes.func.isRequired,
     suspendApp: React.PropTypes.func.isRequired,
-    tasksFetchState: React.PropTypes.number.isRequired
+    tasksFetchState: React.PropTypes.number.isRequired,
+    router: React.PropTypes.object.isRequired
   },
 
   getInitialState: function () {
+    var appid = this.props.model.get("id");
+    var activeTabId;
+
+    var tabs = _.reduce(tabsTemplate, function (current, tab) {
+      var id = tab.id.replace(":appid", appid);
+      if (activeTabId == null) {
+        activeTabId = id;
+      }
+      current.push({
+        id: id,
+        text: tab.text
+      });
+
+      return current;
+    }, []);
+
     return {
       activeViewIndex: 0,
-      activeTabId: tabs[0].id,
-      selectedTasks: {}
+      activeTabId: activeTabId,
+      selectedTasks: {},
+      tabs: tabs
     };
   },
 
+  componentDidMount: function () {
+    this.props.handleSetAppView (function (appid, view) {
+      if (appid && this.isMounted()) {
+        this.setState({
+          activeTabId: "apps/" + appid + (view ? "/" + view : "")
+        });
+      }
+    }.bind(this));
+
+    this.setState({
```

Comment:
I've had quite a few nasty surprises doing this (you'll get some ugly double renders and can occasionally get into a loop).